src/services/anchor-service.ts

```diff
@@ -330,14 +330,16 @@ export class AnchorService {
    */
   async _createIPFSProof(tx: Transaction, merkleRootCid: CID): Promise<CID> {
     const txHashCid = Utils.convertEthHashToCid(tx.txHash.slice(2))
-    const ipfsAnchorProof = {
-      blockNumber: tx.blockNumber,
-      blockTimestamp: tx.blockTimestamp,
-      root: merkleRootCid,
-      chainId: tx.chain,
-      txHash: txHashCid,
-      version: 1,
-    }
+    const ipfsAnchorProof = Object.assign(
+      {
+        blockNumber: tx.blockNumber,
+        blockTimestamp: tx.blockTimestamp,
+        root: merkleRootCid,
+        chainId: tx.chain,
+        txHash: txHashCid,
+      },
+      this.config.useSmartContractAnchors && { version: 1 }
+    )
```

Comment:
```suggestion
    const ipfsAnchorProof = {
      blockNumber: tx.blockNumber,
      blockTimestamp: tx.blockTimestamp,
      root: merkleRootCid,
      chainId: tx.chain,
      txHash: txHashCid,
    }
    if (this.config.useSmartContractAnchors) ipfsAnchorProof.version = 1
```
Can we make this a bit easier to read please?

Reply:
Oh, typescript might not like that?

Reply:
only because the packages need to be updated to the latest version. I can work around